Index: web/core/src/main/java/org/geoserver/web/data/store/AbstractCoverageStorePage.java
===================================================================
--- web/core/src/main/java/org/geoserver/web/data/store/AbstractCoverageStorePage.java	(revision 16422)
+++ web/core/src/main/java/org/geoserver/web/data/store/AbstractCoverageStorePage.java	(working copy)
@@ -57,13 +57,13 @@
         // name
         PropertyModel nameModel = new PropertyModel(model, "name");
         final TextParamPanel namePanel = new TextParamPanel("namePanel", nameModel,
-                new ResourceModel("dataSrcName", "Data Source Name"), true);
+                new ResourceModel("AbstractCoverageStorePage.dataSrcName", "Data Source Name"), true);
 
         paramsForm.add(namePanel);
 
         // description and enabled
         paramsForm.add(new TextParamPanel("descriptionPanel", new PropertyModel(model,
-                "description"), new ResourceModel("description", "Description"), false));
+                "description"), new ResourceModel("AbstractCoverageStorePage.description", "Description"), false));
         paramsForm.add(new CheckBoxParamPanel("enabledPanel", new PropertyModel(model, "enabled"),
                 new ResourceModel("enabled", "Enabled")));
         // a custom converter will turn this into a namespace url
Index: web/core/src/main/java/org/geoserver/web/data/store/AbstractDataAccessPage.java
===================================================================
--- web/core/src/main/java/org/geoserver/web/data/store/AbstractDataAccessPage.java	(revision 16422)
+++ web/core/src/main/java/org/geoserver/web/data/store/AbstractDataAccessPage.java	(working copy)
@@ -103,7 +103,7 @@
 
         {
             final IModel wsModel = new PropertyModel(model, "workspace");
-            final IModel wsLabelModel = new ResourceModel("AbstractDataAccessPage.workspace");
+            final IModel wsLabelModel = new ResourceModel("workspace", "Workspace");
             workspacePanel = new WorkspacePanel("workspacePanel", wsModel, wsLabelModel, true);
         }
         paramsForm.add(workspacePanel);
@@ -116,7 +116,7 @@
         paramsForm.add(dataStoreNamePanel);
 
         paramsForm.add(new TextParamPanel("dataStoreDescriptionPanel", new PropertyModel(model,
-                "description"), new ResourceModel("description", "Description"), false,
+                "description"), new ResourceModel("AbstractDataAccessPage.description", "Description"), false,
                 (IValidator[]) null));
 
         paramsForm.add(new CheckBoxParamPanel("dataStoreEnabledPanel", new PropertyModel(model,
Index: web/core/src/main/java/org/geoserver/web/data/store/AbstractWMSStorePage.java
===================================================================
--- web/core/src/main/java/org/geoserver/web/data/store/AbstractWMSStorePage.java	(revision 16422)
+++ web/core/src/main/java/org/geoserver/web/data/store/AbstractWMSStorePage.java	(working copy)
@@ -51,13 +51,13 @@
         // name
         PropertyModel nameModel = new PropertyModel(model, "name");
         final TextParamPanel namePanel = new TextParamPanel("namePanel", nameModel,
-                new ResourceModel("dataSrcName", "Data Source Name"), true);
+                new ResourceModel("AbstractWMSStorePage.dataSrcName", "Data Source Name"), true);
 
         form.add(namePanel);
 
         // description and enabled
         form.add(new TextParamPanel("descriptionPanel", new PropertyModel(model,
-                "description"), new ResourceModel("description", "Description"), false));
+                "description"), new ResourceModel("AbstractWMSStorePage.description", "Description"), false));
         form.add(new CheckBoxParamPanel("enabledPanel", new PropertyModel(model, "enabled"),
                 new ResourceModel("enabled", "Enabled")));
         // a custom converter will turn this into a namespace url
Index: web/core/src/main/resources/GeoServerApplication.properties
===================================================================
--- web/core/src/main/resources/GeoServerApplication.properties	(revision 16422)
+++ web/core/src/main/resources/GeoServerApplication.properties	(working copy)
@@ -18,12 +18,11 @@
 
 AbstractCoverageStorePage.dataSrcName = Data Source Name
 AbstractCoverageStorePage.description = Description
-AbstractCoverageStorePage.workspace   = Workspace
 
 AbstractDataAccessPage.cantGetDataStoreFactory = Can't find the factory for the given data store. Make sure the \
         appropriate plugin is installed
 AbstractDataAccessPage.dataSrcName             = Data Source Name
-AbstractDataAccessPage.workspace               = Workspace
+AbstractDataAccessPage.description             = Description
 
 AttributeEditPage.title = Edit attribute
 AttributeEditPage.description = Create a new attribute by specifying name, data type and other details
@@ -58,8 +57,7 @@
 
 
 AbstractWMSStorePage.dataSrcName             = WMS Source Name
-AbstractWMSStorePage.description             = WMS cascading store
-AbstractWMSStorePage.workspace               = Workspace
+AbstractWMSStorePage.description             = Description
 AbstractWMSStorePage.capabilitiesURL         = Capabilities URL
 AbstractWMSStorePage.connectionInfo          = Connection Info
 AbstractWMSStorePage.userName                = User Name
