Index: FormatTest.java
===================================================================
--- FormatTest.java	(revision 11156)
+++ FormatTest.java	(working copy)
@@ -8,6 +8,8 @@
 
 import junit.framework.Test;
 
+import org.geoserver.rest.format.MapJSONFormat;
+import org.geoserver.rest.format.MapXMLFormat;
 import org.geoserver.test.GeoServerTestSupport;
 
 /**
@@ -28,8 +30,8 @@
     	super.oneTimeSetUp();
     	// Add to this list if you would like to test another general format
         formats = new ArrayList();
-        formats.add(new AutoXMLFormat());
-        formats.add(new JSONFormat());
+        formats.add(new MapXMLFormat());
+        formats.add(new MapJSONFormat());
     }
 
     public void testFormatMap(){
@@ -40,7 +42,7 @@
                 Map input = new HashMap();
                 input.put("Hello", "Goodbye");
 
-                Map result = (Map)format.readRepresentation(format.makeRepresentation(input));
+                Map result = (Map)format.toObject(format.toRepresentation(input));
                 assertEquals(result.size(), input.size());
                 Iterator mapIt = input.entrySet().iterator();
                 while (mapIt.hasNext()){
@@ -61,7 +63,7 @@
                 List input = new ArrayList();
                 input.add("Hello");
 
-                List result = (List)format.readRepresentation(format.makeRepresentation(input));
+                List result = (List)format.toObject(format.toRepresentation(input));
                 assertEquals(result.size(), input.size());
                 for (int i = 0; i < result.size(); i++){
                     assertEquals(input.get(i), result.get(i)); }
@@ -79,7 +81,7 @@
             DataFormat format = (DataFormat)it.next();
             String input = "Hello";
 
-            String result = (String) format.readRepresentation(format.makeRepresentation(input));
+            String result = (String) format.toObject(format.toRepresentation(input));
             assertEquals(input, result);
         }
     }
